packages/react-native-session-replay/android/src/main/kotlin/com/datadog/reactnative/sessionreplay/mappers/ReactTextMapper.kt

```diff
@@ -0,0 +1,83 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.reactnative.sessionreplay.mappers
+
+import android.widget.TextView
+import androidx.annotation.VisibleForTesting
+import com.datadog.android.sessionreplay.internal.AsyncJobStatusCallback
+import com.datadog.android.sessionreplay.internal.recorder.MappingContext
+import com.datadog.android.sessionreplay.internal.recorder.mapper.TextViewMapper
+import com.datadog.android.sessionreplay.model.MobileSegment
+import com.datadog.reactnative.sessionreplay.ReactTextShadowNodeUtils
+import com.datadog.reactnative.sessionreplay.ReactTextPropertiesResolver
+import com.datadog.reactnative.sessionreplay.extensions.resolveShapeAndBorder
+import com.datadog.reactnative.sessionreplay.getReactBackgroundFromDrawable
+import com.facebook.react.bridge.ReactContext
+import com.facebook.react.views.view.ReactViewBackgroundDrawable
+
+internal class ReactTextMapper(): TextViewMapper() {
+    private lateinit var reactTextPropertiesResolver: ReactTextPropertiesResolver
+
+    internal constructor(
+        reactContext: ReactContext
+    ): this() {
+        val reactTextShadowNodeUtils = ReactTextShadowNodeUtils(reactContext)
+        this.reactTextPropertiesResolver = ReactTextPropertiesResolver(reactTextShadowNodeUtils)
+    }
+
+    internal constructor(
+        reactContext: ReactContext,
+        reactTextShadowNodeUtils: ReactTextShadowNodeUtils = ReactTextShadowNodeUtils(reactContext),
+        reactTextPropertiesResolver: ReactTextPropertiesResolver = ReactTextPropertiesResolver(reactTextShadowNodeUtils)
+    ) : this() {
+        this.reactTextPropertiesResolver = reactTextPropertiesResolver
+    }
+
+    override fun map(
+        view: TextView,
+        mappingContext: MappingContext,
+        asyncJobStatusCallback: AsyncJobStatusCallback
+    ): List<MobileSegment.Wireframe> {
+        val wireframes = mapOnSuperclass(view, mappingContext, asyncJobStatusCallback)
+
+        val result: MutableList<MobileSegment.Wireframe> = mutableListOf()
+        wireframes.forEach{
+            if (it !is MobileSegment.Wireframe.TextWireframe) {
+                result.add(it)
+            } else {
+                val pixelDensity = mappingContext.systemInformation.screenDensity
+
+                // view.alpha is the value of the opacity prop on the js side
+                val opacity = view.alpha
+
+                val backgroundDrawable: ReactViewBackgroundDrawable? =
+                    getReactBackgroundFromDrawable(view.background)
+
+                val (shapeStyle, border) =
+                    backgroundDrawable?.resolveShapeAndBorder(opacity, pixelDensity)
+                        ?: view.background?.resolveShapeStyleAndBorder(opacity)
```

Comment:
Question: In which case would we fall back to the second case? 
Can you add a comment to specify when this happens?

Reply:
This was part of the fallback to textview properties. I'll remove this fallback mechanism, so that we'll always either use shadow properties or null